### ISSUE TITLE
Problems generating update site artifacts

### ISSUE THREAD

**Issue by regrog** [groovy/groovy-eclipse#342]
Eclipse oxygen
java 8

I have and update site plugin, with the latest version (2.9.2.xx-201709250148-e47) when I build the features in it a log file is generated with this error:

> org.eclipse.jdt.internal.compiler.batch.ClasspathJar cannot be cast to java.lang.String

And basically all artifacts are corrupted. If installed in Eclipse, they don't work.
With an old version (2.9.2.xx-20170808) it works.

**Comment by eric-milles** [comment]
Hard to know where to go with this without more details.  Trhe Eclipse Oxygen JDT patch was updated over the weekend.  If that causes compiler errors for you down the line, you would need to address them.  Or you could get the previous update site archive so you are not dependent on snapshot deployments.

**Comment by regrog** [comment]
Well, I don't have any compile errors, everything seems to be created correctly but only after an eclipse installation the error occur.
Eclipse is not able to load the wizard I created.

No useful errors in eclipse log ...

**Comment by eric-milles** [comment]
Can you describe you plug-in or feature a little bit more?  I'm not sure how groovy-eclipse fits in.

**Comment by regrog** [comment]
The fact is that my plugin doesn't involve groovy but jdt.
I think that the problem is the new version of jdt: org.eclipse.jdt.core_3.13.50.xx-201709250148-e47
A fresh oxygen eclipse works fine, after installing groovy-eclipse from http://dist.springsource.org/snapshot/GRECLIPSE/e4.7/ the export doesn't work anymore.

**Comment by eric-milles** [comment]
org.eclipse.jdt.core_3.13.50 is due to be released any day now.  You may want to try out the latest M builds.

**Comment by eric-milles** [comment]
Do you have a stack trace for the cast exception?  It is not clear where the problem is occurring.

**Comment by regrog** [comment]
Sorry but not even in debug mode eclipse produces a stack trace, the only info I have is this:

> org.eclipse.jdt.internal.compiler.batch.ClasspathJar cannot be cast to java.lang.String

I create 3 projects:
myplugin
myFeature
myUpdateSite

try these 2 versions:
1- [testNoLog.zip](https://github.com/groovy/groovy-eclipse/files/1339846/testNoLog.zip)
2- [testWithLog.zip](https://github.com/groovy/groovy-eclipse/files/1339847/testWithLog.zip)

the difference is that (2) has a java class in myplugin
Import the projects in your eclipse Oxygen and try to build from site.xml into myUpdateSite

if there is a java class, the log is produced.

I'm using eclipse oxygen
java 8
org.eclipse.jdt.core jar version is 3.13.50.xx-201709271833-e47
org.codehaus.groovy.jdt.patch version is 2.9.2.xx-201709271833-e47

**Comment by eric-milles** [comment]
One thing you can try is to add some compiler args to your build.properties file:

compilerArg=-nowarn

All supported arguments are detailed here: https://help.eclipse.org/oxygen/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_batch_compiler.htm

**Comment by regrog** [comment]
I tried it but it doesn't work.

The log isn't really about warnings, I'm not interesting in warning suppression.
The log generated by eclipse clearly means that the class ClasspathJar is cast to String, that's not a warning but a real error.

even the last version has the problem, it's related to the jdt.core plugin version.

> Eclipse Compiler for Java(TM) xx-201709282218-e47, 3.13.50, Copyright IBM Corp 2000, 2015. All rights reserved.
> org.eclipse.jdt.internal.compiler.batch.ClasspathJar cannot be cast to java.lang.String

**Comment by eric-milles** [comment]
I wasn't recommending you use the "nowarn" option.  I just was showing how you set compiler/builder options so that the log has more useful information in it.

**Comment by eric-milles** [comment]
Quick update: The log is generated when exporting the plug-in all by itself.  So the feature and update site are not needed for recreation.  Oxygen 4.7.1a RC2 was just released.  I'm going to have a look if that makes a difference.

**Comment by eric-milles** [comment]
The deploy worked in an Oxygen 1a ([RC2](http://download.eclipse.org/eclipse/downloads/drops4/M-4.7.1aRC2-201709281310/)) environment.  Is this something you could update your PDE to?

**Comment by regrog** [comment]
I'm not comfortable to upgrade before release, anyway I gave it a try and it worked fine.
Thanks
